tests/test_cliCmdRetrieveArtifacts.py

```diff
@@ -0,0 +1,85 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Unit tests for daf_butler CLI retrieve-artifacts command.
+"""
+
+import os
+import unittest
+from typing import List
+
+from lsst.daf.butler.cli.butler import cli
+from lsst.daf.butler.cli.utils import clickResultMsg, LogCliRunner
+from lsst.daf.butler import StorageClassFactory, ButlerURI
+from lsst.daf.butler.tests.utils import ButlerTestHelper, makeTestTempDir, MetricTestRepo, removeTestTempDir
+
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class CliRetrieveArtifactsTest(unittest.TestCase, ButlerTestHelper):
+
+    configFile = os.path.join(TESTDIR, "config/basic/butler.yaml")
+    storageClassFactory = StorageClassFactory()
+
+    def setUp(self):
+        self.root = makeTestTempDir(TESTDIR)
+        self.testRepo = MetricTestRepo(self.root,
+                                       configFile=self.configFile)
+
+    def tearDown(self):
+        removeTestTempDir(self.root)
+
+    @staticmethod
+    def find_files(root: str) -> List[ButlerURI]:
+        return list(ButlerURI.findFileResources([root]))
+
+    def testRetrieveAll(self):
+        runner = LogCliRunner()
+        with runner.isolated_filesystem():
+
+            for counter, (preserve_path, prefix) in enumerate((("--preserve-path",
+                                                               "ingest/run/test_metric_comp."),
+                                                              ("--no-preserve-path",
+                                                               "test_metric_comp_"))):
```

Comment:
Why `.` in one and `_` in the other?

Reply:
because the file name uses a `_` to represent the component but the directory is allowed to use a `.` in it. I think what you are really asking is for a comment to say that.